exchange/bitswap/bitswap.go

```diff
@@ -209,17 +206,17 @@ func (bs *Bitswap) GetBlocks(ctx context.Context, keys []key.Key) (<-chan *block
 	}
 
 	bs.wm.WantBlocks(keys)
+	go func() {
+		<-ctx.Done()
+		bs.CancelWants(keys)
```

Comment:
we're going to need ref-counting for this. if another proc asks for them we dont want them cancelled.